File: _/2183OS_08_Code/Task 3/solr_index.rb
require 'rubygems'
require 'rsolr'

solr = RSolr.connect :url => 'http://dev.fts-wizards.com:8983/solr/'

documents = [
	{:id => 1, :name => 'Document 1'}, 
	{:id => 2, :name => 'Document 2'}
]

solr.add documents

solr.update :data => '<commit/>'
solr.update :data => '<optimize/>'

File: _/2183OS_08_Code/Task 2/solr_index.php
<?php
require_once 'Apache/Solr/Service.php';

$solr = new Apache_Solr_Service("dev.fts-wizards.com", "8983", "/solr");

$docs = array(
    'doc1' => array(
      'id' => '1',
      'name' => 'Document one',
    ),
    'doc2' => array(
      'id' => 2,
      'name' => 'Document two',
    ),
  );
    
  $documents = array();
  foreach ( $docs as $item => $fields ) {
    $solrdoc = new Apache_Solr_Document();
    foreach ( $fields as $key => $value ) {
        $solrdoc->$key = $value;
    }
    $documents[] = $solrdoc;
  }
    
  try {
    $solr->addDocuments( $documents );
    $solr->commit();
    $solr->optimize();
  }
  catch ( Exception $ex ) {
    echo $ex->getMessage();
  }
?>